cdap-ui/graphql/graphql.js

```diff
@@ -68,15 +62,33 @@ const resolvers = {
   },
   ProgramRecord: { __resolveType: programsTypeResolver.ProgramRecord.__resolveType },
   Workflow: {
-    runs: runsResolver.Workflow.runs,
+    runs: programsTypeResolver.Workflow.runs,
+    totalRuns: programsTypeResolver.Workflow.totalRuns,
     schedules: schedulesResolver.Workflow.schedules,
   },
+  Spark: {
+    runs: programsTypeResolver.Spark.runs,
+    totalRuns: programsTypeResolver.Spark.totalRuns,
+  },
+  MapReduce: {
+    runs: programsTypeResolver.MapReduce.runs,
+    totalRuns: programsTypeResolver.MapReduce.totalRuns,
+  },
   ScheduleDetail: { nextRuntimes: nextRuntimesResolver.ScheduleDetail.nextRuntimes },
 };
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
+  context: ({ req }) => {
+    const auth = req.headers.authorization;
```

Comment:
headers might not exist, and this will throw an error

Reply:
Updated the check